spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/ConfigMapPropertySourceLocator.java

```diff
@@ -67,7 +68,7 @@ protected abstract MapPropertySource getMapPropertySource(String applicationName
 
 			CompositePropertySource composite = new CompositePropertySource("composite-configmap");
 			if (this.properties.isEnableApi()) {
-				List<NormalizedSource> sources = this.properties.determineSources();
```

Comment:
I am proposing a minor change here. If users have configured (even by accident), a duplicate config map, we should not be reading it twice and create two PropertySources out of that. Ideally, we should be doing that in `ConfigMapConfigProperties`, where we could change `paths` and `sources` to be `Set`s, not `List`s. Of course, that is a breaking change, so the next best thing is to get a `Set` here and thus unique sources.

Reply:
Could this not end up messing with the order of the property sources?  That is important

Reply:
hmm, then I am a bit confused, may be I miss something. Let's suppose this is my config :

```
spring:
  cloud:
    kubernetes:
      config:
        namespace: spring-k8s
        sources:
          - name: config-map-c
          - name: config-map-a
          - name: config-map-a
```

This is the course of actions that we will do:

```
CompositePropertySource composite = new CompositePropertySource("composite-configmap");

 - MapPropertySource source = read "config-map-c"; composite.addFirstPropertySource(source);
 - MapPropertySource source = read "config-map-a"; composite.addFirstPropertySource(source);
 - MapPropertySource source = read "config-map-a"; composite.addFirstPropertySource(source);
```

With the change that I am proposing, it will be : 


```
 - MapPropertySource source = read "config-map-a"; composite.addFirstPropertySource(source);
 - MapPropertySource source = read "config-map-c"; composite.addFirstPropertySource(source);
```

Notice that reading happens only for two of them and they are read in reverse (I did this on purpose here to emphasize my point ).

---

If I get your point correctly, you are saying that if users specify : 

```
- name: config-map-c
- name: config-map-a
```

vs 

```
- name: config-map-a
- name: config-map-c
```

this somehow matters?

Reply:
Yes is does, because the order of the property sources determines which one takes priority when there are conflicting properties.  So if config map a and c have the same property defined if config map c comes last it will override the property from config map a.

Reply:
this is _exactly_ what I thought. You make sense now, thank you.